src/fides/api/alembic/migrations/versions/5fe01e730171_add_external_id_to_privacy_preference.py

```diff
@@ -0,0 +1,269 @@
+            """
+            ALTER TABLE currentprivacypreferencev2
+            ADD CONSTRAINT last_saved_for_external_id_per_property_id
+            UNIQUE USING INDEX ix_currentprivacypreferencev2_external_id_property_id;
+            """
+        )
+        op.execute(
+            """
+            ALTER TABLE currentprivacypreferencev2
+            ADD CONSTRAINT last_saved_for_fides_user_device_per_property_id
+            UNIQUE USING INDEX ix_currentprivacypreferencev2_fides_user_device_property_id;
+            """
+        )
+        op.execute(
+            """
+            ALTER TABLE currentprivacypreferencev2
+            ADD CONSTRAINT last_saved_for_phone_number_per_property_id
+            UNIQUE USING INDEX ix_currentprivacypreferencev2_phone_number_property_id;
+            """
+        )
+    else:
+        logger.warning(
+            "The currentprivacypreferencev2 table has more than 1 million rows, "
+            "skipping index and constraint creation. Be sure to manually run the following commands:\n"
+            "- 'CREATE UNIQUE INDEX CONCURRENTLY ix_currentprivacypreferencev2_email_property_id "
+            "ON currentprivacypreferencev2 (email, property_id)'\n"
+            "- 'CREATE UNIQUE INDEX CONCURRENTLY ix_currentprivacypreferencev2_external_id_property_id "
+            "ON currentprivacypreferencev2 (external_id, property_id)'\n"
+            "- 'CREATE UNIQUE INDEX CONCURRENTLY ix_currentprivacypreferencev2_fides_user_device_property_id "
+            "ON currentprivacypreferencev2 (fides_user_device, property_id)'\n"
+            "- 'CREATE UNIQUE INDEX CONCURRENTLY ix_currentprivacypreferencev2_phone_number_property_id "
+            "ON currentprivacypreferencev2 (phone_number, property_id)'\n"
+            "- 'CREATE INDEX CONCURRENTLY ix_currentprivacypreferencev2_property_id "
+            "ON currentprivacypreferencev2 (property_id)'\n"
+            "- 'CREATE INDEX CONCURRENTLY ix_currentprivacypreferencev2_hashed_external_id "
+            "ON currentprivacypreferencev2 (hashed_external_id)'\n"
+            "- 'ALTER TABLE currentprivacypreferencev2 "
+            "ADD CONSTRAINT last_saved_for_email_per_property_id "
+            "UNIQUE USING INDEX ix_currentprivacypreferencev2_email_property_id'\n"
+            "- 'ALTER TABLE currentprivacypreferencev2 "
+            "ADD CONSTRAINT last_saved_for_external_id_per_property_id "
+            "UNIQUE USING INDEX ix_currentprivacypreferencev2_external_id_property_id'\n"
+            "- 'ALTER TABLE currentprivacypreferencev2 "
+            "ADD CONSTRAINT last_saved_for_fides_user_device_per_property_id "
+            "UNIQUE USING INDEX ix_currentprivacypreferencev2_fides_user_device_property_id'\n"
+            "- 'ALTER TABLE currentprivacypreferencev2 "
+            "ADD CONSTRAINT last_saved_for_phone_number_per_property_id "
+            "UNIQUE USING INDEX ix_currentprivacypreferencev2_phone_number_property_id'"
+        )
```

Comment:
Because this is non-standard, we haven't had an instance of this in our codebase, can you add code comments to the migration particularly around how you're creating the unique index concurrently, and then using that index that is already unique to create a constraint and why to help future devs?

Reply:
Added this comment to the top of the migration
```
WARNING - Conditional migration based on table size

This migration script checks the size of specific tables and performs the following actions:

1. If the tables have less than 1 million rows:
   - Creates the necessary indices
   - Adds the required constraints

2. If the tables are too large (1 million rows or more):
   - Logs non-blocking SQL commands to create the indices and constraints
   - The non-blocking commands utilize:
     - `CREATE INDEX CONCURRENTLY` for creating indices
     - `ADD CONSTRAINT USING INDEX` for adding constraints
   - These commands add the indices and constraints without blocking any reads on the tables
```

Reply:
nice this looks great -